vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewModel.kt

```diff
@@ -43,21 +45,42 @@ class SessionOverviewViewModel @AssistedInject constructor(
     }
 
     init {
-        val currentDeviceId = session.sessionParams.deviceId.orEmpty()
         setState {
-            copy(isCurrentSession = deviceId.isNotEmpty() && deviceId == currentDeviceId)
+            copy(isCurrentSession = isCurrentSession(deviceId))
         }
-
         observeSessionInfo(initialState.deviceId)
     }
 
+    private fun isCurrentSession(deviceId: String): Boolean {
+        return isCurrentSessionUseCase.execute(deviceId)
+    }
+
     private fun observeSessionInfo(deviceId: String) {
         getDeviceFullInfoUseCase.execute(deviceId)
                 .onEach { setState { copy(deviceInfo = Success(it)) } }
                 .launchIn(viewModelScope)
     }
 
     override fun handle(action: SessionOverviewAction) {
-        TODO("Implement when adding the first action")
+        when (action) {
+            is SessionOverviewAction.VerifySession -> handleVerifySessionAction()
+        }
+    }
+
+    private fun handleVerifySessionAction() = withState { viewState ->
+        if (isCurrentSession(viewState.deviceId)) {
```

Comment:
You could use the value which is in the state:
```suggestion
        if (viewState.isCurrentSession) {
```

Also there is not else block? Maybe handled in another PR? I would add a // TODO if it's the case, to not forget.
I'm scared of having a button with no effect, but maybe this is not possible (did not check).

Reply:
Yes, you are right but no need for a TODO as it it already implemented in the following PR #7156 which will be merged just right after this one.

Reply:
And for the suggestion, I will do the update in the following PR to avoid blocking this one.